controllers/fenceagentsremediation_controller.go

```diff
@@ -70,69 +69,54 @@ type FenceAgentsRemediationReconciler struct {
-		r.Log.Error(err, "failed to get FAR")
-		return ctrl.Result{}, err
+		r.Log.Error(err, "failed to get FAR CR")
+		return emptyResult, err
 	}
-	key := client.ObjectKey{Namespace: req.Namespace, Name: fenceAgentsTemplateName}
-	farTemplate := &v1alpha1.FenceAgentsRemediationTemplate{}
-	if err := r.Get(ctx, key, farTemplate); err != nil {
-		r.Log.Error(err, "failed to get FAR template")
-		return ctrl.Result{}, err
-	}
-
-	pod, err := r.getFAPod(req.NamespacedName.Namespace)
+	// TODO: Validate FAR CR name to nodeName. Run isNodeNameValid
+	// Fetch the FAR's pod
+	r.Log.Info("Fetch FAR's pod")
+	pod, err := r.getFenceAgentsPod(req.Namespace)
 	if err != nil {
-		return ctrl.Result{}, err
+		return emptyResult, err
 	}
 
+	// Build CLI executer for FAR's pod
+	r.Log.Info("Build CLI executer for FAR's pod")
 	ex, err := cli.NewExecuter(pod)
 	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	faParams := buildFenceAgentParams(farTemplate, far)
-	cmd := append([]string{farTemplate.Spec.Agent}, faParams...)
-	//fence_ipmilan --ip=192.168.111.1 --ipport=6233 --username=admin --password=password --action=status --lanplus --verbose
-	if _, _, err := ex.Execute(cmd); err != nil {
-		return ctrl.Result{}, err
+		return emptyResult, err
 	}
 
-	return ctrl.Result{}, nil
-}
-
-func buildFenceAgentParams(farTemplate *v1alpha1.FenceAgentsRemediationTemplate, far *v1alpha1.FenceAgentsRemediation) []string {
-	var fenceAgentParams []string
-	for paramName, paramVal := range farTemplate.Spec.SharedParameters {
-		fenceAgentParams = appendParamToSlice(fenceAgentParams, string(paramName), paramVal)
-
+	//TODO: Check that FA is excutable? run cli.IsExecuteable
+	r.Log.Info("Create and execute the fence agent", "Fence Agent", far.Spec.Agent)
+	faParams, err := buildFenceAgentParams(far)
+	if err != nil {
+		return emptyResult, err
 	}
-
-	nodeName := v1alpha1.NodeName(far.Name)
-	for paramName, nodeMap := range farTemplate.Spec.NodeParameters {
-		fenceAgentParams = appendParamToSlice(fenceAgentParams, string(paramName), nodeMap[nodeName])
+	cmd := append([]string{far.Spec.Agent}, faParams...)
+	// The Fence Agent is excutable and the parameters are valid but we don't know about their values
+	if _, _, err := ex.Execute(cmd); err != nil {
+		//TODO: better seperation between errors from wrong shared parameters values and wrong node parameters values
+		return emptyResult, err
 	}
 
-	return fenceAgentParams
-}
-
-// SetupWithManager sets up the controller with the Manager.
-func (r *FenceAgentsRemediationReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.FenceAgentsRemediation{}).
-		Complete(r)
+	return emptyResult, nil
 }
 
-func (r *FenceAgentsRemediationReconciler) getFAPod(namespace string) (*corev1.Pod, error) {
+// getFenceAgentsPod fetches the FAR pod based on FAR's label and namespace
+func (r *FenceAgentsRemediationReconciler) getFenceAgentsPod(namespace string) (*corev1.Pod, error) {
```

Comment:
something for a follow up, not this PR: you don't need to search the pod, you can know its namespace and name, see here for how to get the namespace, same should work for the name: https://github.com/medik8s/node-healthcheck-operator/blob/main/config/manager/manager.yaml#L53

Reply:
Good point, and it sounds like a better coding